blogposts/2022/css-modules-migration.md

```diff
@@ -0,0 +1,255 @@
+
+1. Using a combination of [Bootstrap utility classes](https://getbootstrap.com/docs/4.0/layout/utilities-for-layout/).
+2. Using a custom class specific to the element.
+3. Using global styles inherited from the global scope.
+
+It was hard to change individual UI components because engineers needed to keep in mind the whole stylesheet to avoid colliding styles in the global scope:
+
+1. When adding a new class name to the stylesheet, engineers needed to ensure no other component already used it. [The BEM methodology](https://en.bem.info/methodology/) helped with this a lot, but we still bumped into issues from time to time. These issues were incredibly difficult to debug as global styles can be used in so many different ways.
+2. Global CSS rules depend on the order of the code, and it's difficult to ensure that the order is preserved when implementing global changes like the application redesign. Moving files around could result in a slightly different import order that affects a tiny bit of UI. Debugging such small changes can significantly slow teams down and makes DX a nightmare.
+3. There was no robust and configured way to identify unused CSS code not linked to any HTML elements. In the process of the redesign, we occasionally uncovered such styles. These styles added no value to our codebase or our product.
+
+To overcome these issues, we decided that, before diving into the redesign, we should find a way to embrace scoped styles instead of adding new styles to the global scope. At the same time, we needed to ensure that this solution could be integrated with the current approach effectively. The definition of success was a combination of the following outcomes:
+
+1. Modular CSS is enforced for new design system components.
+2. It works well with existing global styles and UI elements.
+3. We have a clear migration path that doesn’t interfere with feature development.
+
+## CSS Modules solution
+
+We researched popular solutions available in the open source universe: trade-offs between CSS in JS, regular CSS, and hybrid solutions. Some offered more powerful features, but we decided to use a solution that we knew we could easily adopt — CSS modules.
+
+### What are CSS Modules?
+
+CSS modules are great because they help you write reusable components with isolated styles. According to the [repo](https://github.com/css-modules/css-modules), CSS modules are:
+
+“CSS files in which all class names and animation names are scoped locally by default.”
+
+With CSS modules, CSS classes should be referenced in JavaScript file via explicit binding to the styles file:
```

Comment:
```suggestion
With CSS modules, CSS classes should be referenced in the JavaScript file via explicit binding to the styles file:
```
This might not be quite right but we're definitely missing a word here! @valerybugakov